rosidl_typesupport_fastrtps_c/test/test_cli_extension.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+from rosidl_cli.command.generate.api import generate
+
+TEST_DIR = str(pathlib.Path(__file__).parent)
+
+
+def test_cli_extension_for_smoke(tmp_path):
+    assert len(generate(
+        package_name='rosidl_typesupport_fastrtps_c',
+        interface_files=[TEST_DIR + ':msg/Test.msg'],
```

Comment:
```suggestion
        interface_files=[TEST_DIR + '/msg/Test.msg'],
```

?

Reply:
If `TEST_DIR` would be a pathlib.Path, you could instead use:

```suggestion
        interface_files=[str(TEST_DIR / 'msg' / 'Test.msg')],
```

which IMO is a bit better

Reply:
About the semicolon (`':'`), it has to be a semicolon. See [here](https://github.com/ros2/rosidl/blob/1d7b22f291aba4a935a45effacb5ca8ce07bf6d3/rosidl_cli/rosidl_cli/command/generate/api.py#L38-L40).

About the trailing relative path, it is POSIX to match the behavior of the pure CMake pipeline. Arguably, the way it is handled inside `rosidl` is a bit mixed up and it'd _probably_ work if it used backslashes on Windows instead. Haven't tried.

Reply:
I don't understand why we're using a semicolon in a path, but if that's the current way... then okay.